pkg/clusters/addons/kong/builder.go

```diff
@@ -36,6 +41,34 @@ func (b *Builder) WithPostgreSQL() *Builder {
 	return b
 }
 
+// WithEnterprise deploying kong enterpise
+func (b *Builder) WithEnterprise() *Builder {
+	b.enterprise = true
+	b.repo = DefaultEnterpriseImageRepo
+	b.tag = DefaultEnterpriseImageTag
+	b.license = KongEnterpriseTestLicense
+	b.kongAdminPassword = EnterpriseKongAdminDefaultPWD
```

Comment:
Setting all these without conditionals in this method slightly breaks this `Builder` API because these attributes are set by other `With<Feature>()` methods.

We should either remove these defaults and place them below in the `Build()` method:

```suggestion
```

OR we should make sure not to implicitly override any values which have already been provided by the caller:

```suggestion
	if b.repo == "" {
	    b.repo = DefaultEnterpriseImageRepo
	}
	if b.tag == "" {
	    b.tag = DefaultEnterpriseImageTag
	}
	if b.license == "" {
	    b.license = KongEnterpriseTestLicense
	}
	if b.kongAdminPassword == "" {
	    b.kongAdminPassword = EnterpriseKongAdminDefaultPWD
	}
```

Making this change will make it possible to use the `Builder` object configuration methods as they were originally intended: without worrying about order of operations.

Reply:
adopted the second one
https://github.com/Kong/kubernetes-testing-framework/pull/104/files#diff-7cc1dbd395c3c0ea6e62a7d5a3d4493f37054d7b7be23f198340cdd6a52c0be2R47-R54